be1-go/db/sqlite_test.go

```diff
@@ -0,0 +1,183 @@
+package db
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"student20_pop/message"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSQLite_NewRepository(t *testing.T) {
+	repo, err := createRepo("test.db")
+	require.NoError(t, err)
+
+	err = repo.Close()
+	require.NoError(t, err)
+}
+
+func TestSQLite_AddMessage(t *testing.T) {
+	repo, err := createRepo("add_message.db")
+	require.NoError(t, err)
+
+	defer repo.Close()
+
+	channelID := "12345"
+	timestamp := message.Timestamp(time.Now().UnixNano())
+
+	msg, err := createMessage(0, timestamp)
+	require.NoError(t, err)
+
+	err = repo.AddMessage(channelID, msg, timestamp)
+	require.NoError(t, err)
+}
+
+func createMessage(i int, timestamp message.Timestamp) (message.Message, error) {
+	data, err := message.NewCreateLAOData("test", timestamp, []byte{1, 2, 3, byte(i)}, []message.PublicKey{})
+	if err != nil {
+		return message.Message{}, err
+	}
+
+	msg := message.Message{
+		MessageID:         []byte{6, 7, 8, 9, byte(i)},
+		Data:              data,
+		Sender:            []byte{1, 2, 3, byte(i)},
+		Signature:         []byte{1, 2, 3, byte(i)},
+		WitnessSignatures: []message.PublicKeySignaturePair{},
+	}
+
+	return msg, nil
+}
+
+func addMessages(repo Repository, channelID string, limit int) (int64, error) {
+	now := time.Now().UnixNano()
+
+	for i := 0; i < limit; i++ {
+		timestamp := message.Timestamp(now + int64(i))
+
+		msg, err := createMessage(i, timestamp)
+		if err != nil {
+			return 0, err
+		}
+
+		err = repo.AddMessage(channelID, msg, timestamp)
+		if err != nil {
+			return 0, err
+		}
+	}
+
+	return now, nil
+}
+
+func TestSQLite_GetMessages(t *testing.T) {
+	repo, err := createRepo("get_messages.db")
+	require.NoError(t, err)
+
+	defer repo.Close()
+
+	channelID := "12345"
+
+	_, err = addMessages(repo, channelID, 100)
+	require.NoError(t, err)
+
+	messages, err := repo.GetMessages(channelID)
+	require.NoError(t, err)
+
+	require.Equal(t, 100, len(messages))
```

Comment:
I don't think it's important at all but would it be a good idea to also check if the messages are in ascending order of their timestamp?

Reply:
Thanks for suggesting that! Updated the tests :)